### PR TITLE
redirect to original request url

### PR DESCRIPTION
**Pull request by banukutlu**
Issue #236

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -37,6 +37,8 @@ def patron_info_response
     end
 
     def authenticate_user!
+      session[:original_fullpath] = request.original_fullpath unless request.original_fullpath == '/'
```
Comment (cdmo) [inline app/controllers/application_controller.rb:40]: 😎  https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-original_fullpath

File: spec/controllers/sessions_controller_spec.rb [hunk 2]
```diff
@@ -22,6 +22,16 @@
       it 'redirects to the home page' do
         expect(get(:index)).to redirect_to summaries_url
       end
+
+      context 'when user requested a page before authentication' do
```
Comment (cdmo) [inline spec/controllers/sessions_controller_spec.rb:26]: I tried a fresh browser request of localhost/holds and it redirected to the homepage after authentication. Is that expected behavior? It seems, if I'm reading the logic correctly that unless my request is for the root it should send me to my requested for destination. Right?

**Review by cdmo — COMMENTED** [review]
Just messing around with this, I gave a shot to just altering `ApplicationController::authenticate_webaccess` to be the following:

```rb
  def authenticate_webaccess
    redirect_to Settings.symws.webaccess_url + request.original_url
  end
```

And it seems to do the trick for all requests and I think it's the only change needed? Did you try this and run into something?

File: spec/controllers/sessions_controller_spec.rb (hunk 2 above)
Comment (banukutlu) [reply spec/controllers/sessions_controller_spec.rb:26]: right, It should be redirecting to `/holds`. you are reading correctly, not sure why it did not work.

**Comment by banukutlu** [comment]
> Just messing around with this, I gave a shot to just altering `ApplicationController::authenticate_webaccess` to be the following:
> 
> ```ruby
>   def authenticate_webaccess
>     redirect_to Settings.symws.webaccess_url + request.original_url
>   end
> ```
> 
> And it seems to do the trick for all requests and I think it's the only change needed? Did you try this and run into something?

No, I haven't tried this one. It looks nicer with less changes 👍

**Comment by banukutlu** [comment]
> ```ruby
>  + request.original_url
> ```

Just tried it but I could not get it to work. Should we chat/zoom tomorrow for this PR?

File: spec/controllers/sessions_controller_spec.rb (hunk 2 above)
Comment (cdmo) [reply spec/controllers/sessions_controller_spec.rb:26]: we good

**Review by cdmo — CHANGES_REQUESTED** [review]
feature spec, as per zoom session chat

**Comment by cdmo** [comment]
the feature spec would have been ok I think, but what you have is shorter and clearer 👍

**Review by cdmo — APPROVED** [review]

**Pull request merged**